src/tree-sitter-language-mode.js

```diff
@@ -1293,18 +1346,20 @@ class NodeRangeSet {
       let position = node.startPosition;
       let index = node.startIndex;
 
-      for (const child of node.children) {
-        const nextIndex = child.startIndex;
-        if (nextIndex > index) {
-          this._pushRange(buffer, previousRanges, result, {
-            startIndex: index,
-            endIndex: nextIndex,
-            startPosition: position,
-            endPosition: child.startPosition
-          });
+      if (!this.includeChildren) {
+        for (const child of node.children) {
+          const nextIndex = child.startIndex;
+          if (nextIndex > index) {
+            this._pushRange(buffer, previousRanges, result, {
+              startIndex: index,
+              endIndex: nextIndex,
+              startPosition: position,
+              endPosition: child.startPosition
+            });
+          }
+          position = child.endPosition;
+          index = child.endIndex;
```

Comment:
This is where we would subtract out the ranges of child nodes when computing injection ranges. The `includeChildren` flag suppresses that behavior.